code/game/machinery/constructable_frame.dm

```diff
@@ -445,6 +445,45 @@ to destroy them and players will be able to make replacements.
 	origin_tech = "programming=1"
 	req_components = list(
 							/obj/item/weapon/stock_parts/matter_bin = 1)
+	var/list/fridge_names_paths = list(
+							"\improper SmartFridge" = /obj/machinery/smartfridge,
+							"\improper MegaSeed Servitor" = /obj/machinery/smartfridge/seeds,
+							"\improper Refrigerated Medicine Storage" = /obj/machinery/smartfridge/medbay,
+							"\improper Slime Extract Storage" = /obj/machinery/smartfridge/secure/extract,
+							"\improper Secure Refrigerated Medicine Storage" = /obj/machinery/smartfridge/secure/medbay,
+							"\improper Smart Chemical Storage" = /obj/machinery/smartfridge/secure/chemistry,
+							"smart virus storage" = /obj/machinery/smartfridge/secure/chemistry/virology,
+							"\improper Drink Showcase" = /obj/machinery/smartfridge/drinks
+	)
+
+
+/obj/item/weapon/circuitboard/smartfridge/New()
+	..()
+	if(build_path == /obj/machinery/smartfridge)			//If we spawn the base type board (determined by the base type machine as the build path), become a random fridge board
+		var/new_path = fridge_names_paths[pick(fridge_names_paths)]
+		set_type(new_path)
+
+
+/obj/item/weapon/circuitboard/smartfridge/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/screwdriver))
+		set_type(null, user)
+
+/obj/item/weapon/circuitboard/smartfridge/proc/set_type(typepath, mob/user)

```

Comment:
I wouldn't be opposed to making the set_type proc more of a circuitboard level proc, except as it stands with this PR, only 3 boards use it and one of those (vendors) then changes its build components, so not sure how much that would really benefit us as of this time.